include/dicom/dicom.h

```diff
@@ -1479,91 +1635,167 @@ DcmFilehandle *dcm_filehandle_create_from_memory(DcmError **error,
                                                  int64_t length);
 
 /**
- * Read File Meta Information from a File.
+ * Destroy a Filehandle.
+ *
+ * :param filehandle: File
+ */
+DCM_EXTERN
+void dcm_filehandle_destroy(DcmFilehandle *filehandle);
+
+/**
+ * Get File Meta Information from a File.
  *
- * Keeps track of the offset of the Data Set relative to the beginning of the
- * filehandle to speed up subsequent access, and determines the Transfer
- * Syntax in which the contained Data Set is encoded.
+ * Reads the File Meta Information and saves it in the File handle. Returns a
+ * reference to this internal copy of the File Meta Information.
+ *
+ * The return result must not be destroyed. Make a clone of it with
+ * :c:func:`dcm_dataset_clone()` if you need it to remain valid after
+ * closing the File handle.
+ *
+ * After calling this function, the filehandle read point is always
+ * positioned at the start of the File metadata.
+ *
+ * It is safe to call this function many times.
  *
  * :param error: Pointer to error object
  * :param filehandle: Pointer to file handle
  *
  * :return: File Meta Information
  */
 DCM_EXTERN
-DcmDataSet *dcm_filehandle_read_file_meta(DcmError **error,
-                                          DcmFilehandle *filehandle);
+const DcmDataSet *dcm_filehandle_get_file_meta(DcmError **error,
+                                               DcmFilehandle *filehandle);
+
+/**
+ * Get Transfer Syntax UID for a fileahndle.
+ *
+ * :param filehandle: File
+ *
+ * :return: UID of the transfer syntax for this File.
+ */
+DCM_EXTERN
+const char *dcm_filehandle_get_transfer_syntax_uid(const DcmFilehandle *filehandle);
 
 /**
  * Read metadata from a File.
  *
- * Keeps track of the offset of the Pixel Data Element relative to the
- * beginning of the filehandle to speed up subsequent access to individual
- * Frame items.
+ * Read slide metadata, stopping when one of the tags in the stop list is
+ * seen. If the stop list pointer is NULL, it will stop on any of the pixel
+ * data tags.
+ *
+ * The return result must be destroyed with :c:func:`dcm_dataset_destroy()`.
+ *
+ * After calling this function, the filehandle read point is always
+ * positioned at the tag that stopped the read. You can call this function
+ * again with a different stop set to read more of the metadata.
  *
  * :param error: Pointer to error object
  * :param filehandle: File
+ * :param stop_tags: NULL, or Zero-terminated array of tags to stop on
```

Comment:
It would be nicer to make it `skip_tags` to skip over certain elements rather than stopping at them. Note that Pixel Data and friends have tags with very high numbers so that they by design always end up at the end of the dataset. Therefore, it makes sense to stop at them. For all other bulkdata, it would be better to skip rather than stop.

Reply:
Unfortunately it's not possible to skip tags in dicom. Most compound tags don't have a length element, so you have to parse the whole thing.